### ISSUE TITLE
Rework scoreboard player count

### ISSUE THREAD

**Issue by sepulzera** [teeworlds/teeworlds#1839]
This issue was split from #1826 for better readability.

> Imo the team player count should be reworked to make it more evident.
>
>![49052311-e860db80-f1b0-11e8-903b-25eddc162255](https://user-images.githubusercontent.com/43857716/49723734-da5a9280-fc67-11e8-80c4-d37d86e545a0.png)

As @1stDecadeTW said:
> the "(0)" isn't informative and the team score must be prioritized.

**Comment by Fisico** [comment]
@sepulzera why you made a new issue? 
i think my "refine the scoreboard" issue is a good place to come up with new ideas for the scoreboard.

**Comment by Zatline** [comment]
Maybe get rid of the counter? That would make it more clean and simple highly prioritized KISS.

**Comment by sepulzera** [comment]
They are two separate concerns, imo. Didn't want to mix them up, to make it easier to to follow those separate discussions.

**Comment by Fisico** [comment]
Highlight the score maybe?
![image](https://user-images.githubusercontent.com/329471/49753107-a5bff880-fcb2-11e8-943c-8fcca62620f0.png)

**Comment by Dune-jr** [comment]
That's discussed in #1826 already.

I agree with @Zatline here, if anything, scrap the counters.

**Comment by sepulzera** [comment]
Removing the counter isn't more KISS at all. In fact, I would even say it is counter-KISS, having to count myself.
BUT I would rather have it dropped for now, until we found a better way.

**Comment by 1stDecadeTW** [comment]
A simple player count isn't enough? 
![screenshot_1](https://user-images.githubusercontent.com/44511917/49848615-8ea90580-fd9b-11e8-905d-94afa0c47382.jpg)

**Comment by 1stDecadeTW** [comment]
What about this options? 

![screenshot_1](https://user-images.githubusercontent.com/44511917/49852521-f9613d80-fda9-11e8-9708-e076412b8215.jpg)
![screenshot_2](https://user-images.githubusercontent.com/44511917/49852522-f9613d80-fda9-11e8-8c0a-cf98d684dd9c.jpg)
![screenshot_7](https://user-images.githubusercontent.com/44511917/49852776-b0f64f80-fdaa-11e8-887e-05fbbbc4b31c.jpg)

![screenshot_3](https://user-images.githubusercontent.com/44511917/49852523-f9613d80-fda9-11e8-9f55-2f5fe037aa42.jpg)
![screenshot_4](https://user-images.githubusercontent.com/44511917/49852524-f9f9d400-fda9-11e8-9e9a-834ce744368e.jpg)
![screenshot_5](https://user-images.githubusercontent.com/44511917/49852525-f9f9d400-fda9-11e8-9f35-6b8b51ff705c.jpg)
![screenshot_6](https://user-images.githubusercontent.com/44511917/49852527-fa926a80-fda9-11e8-9890-34fc1ef3f2cb.jpg)

**Comment by Zatline** [comment]
> What about this options?
> 

> ![screenshot_4](https://user-images.githubusercontent.com/44511917/49852524-f9f9d400-fda9-11e8-9e9a-834ce744368e.jpg)

this one is awesome!

**Comment by sepulzera** [comment]
The per team player counts are too small. I assume they are very hard to read on lower resolutions.
But I like the overall count, though you should add a colon after "Match" and "Players", as there is one after "Score limit" and "Spectators".

**Comment by Dune-jr** [comment]
Agree, it's a small font (and there is no blur in Teeworlds). Adding more height to the scoreboard isn't great, there isn't a lot of wiggle room.

I don't see what's wrong with the current counters, really. They're grayed out, they don't catch too much attention.

**Comment by sepulzera** [comment]
Putting numbers without any label is not self-explanatory.

Just another brainstorm: 
Do we display dead/alive in the scoreboard?
This could be especially nice to have for survival gametype.
In CS:GO the score line of dead players are grayed out. And the team player counter would be updated to f. e. "Alive 4/5".

**Comment by Dune-jr** [comment]
There is already "Spectators (7)" and it's just fine. You don't need a label to understand it means 7 players.

Same goes for "Red Team (7)"

**Comment by sepulzera** [comment]
For spectators it works because it is a simple list. But the ingame score information is much more complex. You only know that Red Team (7) means 7 players from experience.
Plus, tbh it looks... 'experimental'. I really thought we could agree on that after all.

**Comment by 1stDecadeTW** [comment]
As @sepulzera said, the "7 players" could be update to"4/7 players alive" in some mods like last man standing, it is a simple way to give the user a personalized and nice game experience.

**Comment by 1stDecadeTW** [comment]
Bigger font player count + mockups of team last man standing scoreboard:

Keep in mind that in lms and similar gametypes, the kills/deaths should be more visible, and the score is not as important as the time you survive. "Survive is a matter of time".

![screenshot_1](https://user-images.githubusercontent.com/44511917/49892159-e7b18180-fe0d-11e8-8737-e06127fae595.jpg)
![screenshot_2](https://user-images.githubusercontent.com/44511917/49892160-e7b18180-fe0d-11e8-9573-87e8ba874243.jpg)
![screenshot_3](https://user-images.githubusercontent.com/44511917/49892161-e84a1800-fe0d-11e8-80d9-bee4910077a8.jpg)

**Comment by Fisico** [comment]
Nice work, cool concepts... pretty similar to mine... I would put the blue teamscore on the right side... so all scores are in one column. The blur is nice but it's better to remove it cause, we can’t do this in teeworlds atm. Without blur I think the scoreboard is to light.

**Comment by 1stDecadeTW** [comment]
@Fisico I used your concept to only show where the player count should be placed, and when I was working on that, I realised that when I'm playing a ctf I have to look to every corner of the scoreboard to compare team scores... So I searched for some game scoreboards and found that all team scores are at the center, so you can quickly and easily compare them.

Just a good example: 
![screenshot_1](https://user-images.githubusercontent.com/44511917/49894169-e6cf1e80-fe12-11e8-9de9-fe882e9d39c6.jpg)

**Comment by sepulzera** [comment]
@1stDecadeTW I love the visualization of the dead players. But on second thought, this feature should only be enabled for gametypes without respawn.

However, I am afraid that the per team player counter won't work at all at this position. The transparency is too high for the small font. I only see two options here:
1. Move them to the top header. This will probably look very ugly.
2. Scrap them.

**Comment by Fisico** [comment]
why not take the header from tf2... the scorebaord has cool stuff..

call the teams "Blue" and "Red" and show the player count as "x players"

**Comment by Fisico** [comment]
tf2 inspired me to "design" a statboard for the player...
![image](https://user-images.githubusercontent.com/329471/49896325-4cdb9600-fe53-11e8-9f2b-92ccccec0cf5.png)

**Comment by 1stDecadeTW** [comment]
It looks good, I like it.

![screenshot_2](https://user-images.githubusercontent.com/44511917/49896989-7fb56800-fe1a-11e8-9fd7-162a88880575.jpg)
![screenshot_3](https://user-images.githubusercontent.com/44511917/49896990-7fb56800-fe1a-11e8-9d9d-dd3382ea9101.jpg)

**Comment by 1stDecadeTW** [comment]
No capslock: 
![screenshot_2](https://user-images.githubusercontent.com/44511917/49897342-7ed10600-fe1b-11e8-9de0-657d77df3566.jpg)
![screenshot_3](https://user-images.githubusercontent.com/44511917/49897343-7ed10600-fe1b-11e8-8d50-d58aadd92e7b.jpg)

**Comment by sepulzera** [comment]
Just played my first match survival. The scoreboard shows the dead tees and below the usual flag and teamscore thingy on the ingame screen (lower right), there is the alive counter printed.

I don't think we need another alive player counter in the scoreboard.

Imo, add the overall ingame player counter to above the Spectators list and it is very fine.

**Comment by 1stDecadeTW** [comment]
I agree, all vanilla gametypes have a max of 16 players, 8 vs 8, it's a number that you can easily keep un your memory.

**Comment by sepulzera** [comment]
Do we all agree to add the overall player counter like @1stDecadeTW suggested?
@oy What do you think, could we go for it for the next milestone 0.7.2?

![playercounter](https://user-images.githubusercontent.com/43857716/49967021-f9844900-ff21-11e8-9e05-cdeb9fc8943a.jpg)

**Comment by oy** [comment]
That space where you put "Players 14/16" is the space for the time limit.

**Comment by 1stDecadeTW** [comment]
In that case, this is where player count should be placed:

![screenshot_2](https://user-images.githubusercontent.com/44511917/50198132-e2719c80-030f-11e9-9da0-b3c1f657578f.jpg)

**Comment by sepulzera** [comment]
Imo the timelimit should be next to scorelimit.

**Comment by 1stDecadeTW** [comment]
> Imo the timelimit should be next to scorelimit (if we go for this).

Not all servers have time limit, and that empty space would look ugly, so the solution I propose is to show a "---" when time limit is disabled.

![screenshot_1](https://user-images.githubusercontent.com/44511917/50238232-70876a80-0384-11e9-855e-c72117646f2f.jpg)
